filesmanager/processors/xapi/event_transformers/filesmanager_events.py

```diff
@@ -0,0 +1,39 @@
+"""
+Transformers for filesmanager events.
+
+Classes:
+    FilesDownloadedTransformer: Transformer for the event edunext.xblock.filesmanager.files.downloaded.
+"""
+
+from tincan import Activity, ActivityDefinition, LanguageMap, Verb
+
+from filesmanager.edxapp_wrapper.event_routing_backends import XApiTransformer, XApiTransformersRegistry
+from filesmanager.processors.xapi import constants
+
+
+@XApiTransformersRegistry.register("edunext.xblock.filesmanager.files.downloaded")
+class FilesDownloadedTransformer(XApiTransformer):
+    """
+    Transformers for event generated when an student download files from xblock.
+    """
+
+    _verb = Verb(
+        id=constants.XAPI_VERB_DOWNLOADED,
+        display=LanguageMap({constants.EN: constants.DOWNLOADED}),
+    )
+
+    additional_fields = ("result",)
+
+    def get_object(self):
+        """
+        Get object for xAPI transformed event related to files download from xblock.
+
+        Returns:
+            `Activity`
+        """
+        return Activity(
+            id=self.get_object_iri("xblock", self.get_data("data.block_id", True)),
```

Comment:
`data.something` makes reference to a field inside the event, so this line is looking for the `block_id` field, however the event doesn't have any` block_id`, there are `course_id` and `xblock_id `.

IMO this should be the S3 or the static url since this is the identifier of the object which is the downloaded file in this case

Reply:
> `data.something` makes reference to a field inside the event, so this line is looking for the `block_id` field, however the event doesn't have any` block_id`, there are `course_id` and `xblock_id `.

Okay, it was not clear to me, thanks!

> IMO this should be the S3 or the static url since this is the identifier of the object which is the downloaded file in this case

Yes, you are right. The problem I found is that the zip file that is downloaded is generated in runtime, then I don't know what value the id could have. Does it make sense for the id to be a list of URLs for each file?

Reply:
that means that the user can download multiple files at the same time ??

Reply:
Yes, the student can choose to download one file or download multiple files. In case want to download multiple files, or maybe download a folder, a .zip file is generated in runtime.

Reply:
I'm not an expert of XApi but I think the list of ids is not possible,  my advice is to emit one event per file

Reply:
@andrey-canon, talking to @Ian2012, told me that it's only important to know that X student downloaded something for the event, and thus have the download metrics. That's why I would leave the xblock_id as the id.

Reply:
Probably that arguments is just valid in your context, and depends on what you want to achieve, if you want to measure the amount of files per xblock it's ok, however if you have an implementation where you are using the xblock with two files (lecture and answers)  and the client wants to know how many students downloaded the answers sheet without downloading the lecture file, this won't be valid or a simple case the client just wants to measure the amount of a specific file... My recommendation is to think out of the box  and consider more implementation cases since this a public repository however if this xblock is designed for specific client  and they just need that keep the simple implementation

Reply:
For the moment, we will handle the event like this, but we will take into account what you say in case it is necessary to have a more detailed event. Thanks for the review, it was a great help!

BTW, [this error](https://github.com/eduNEXT/xblock-filesmanager/pull/35#issuecomment-1872293211) was happening because the `event/` endpoint was not sending the `courserun_key`, when it was sent it was solved.